### PR TITLE
Collimation result, redesign the interface of gather module

### PR DESCRIPTION
**Pull request by luxiaohan**
This PR aims to add new interface to parse the results of collimation study. 
For the moment, the results from collimation study are `aperture_losses.dat`, `Coll_Scatter.dat`, `initial_state.dat`, `final_state.dat`. 
Due to more result files are involved, the original interface of gather module is out of date.
So I changed the interface to make it more flexible, the new interface is easier to add new method to parse new files.
At the same time, I move the definations of database table to a seperate module `dbtable.py` to make it more readable. 
Also fixed some bugs.

### CODE REVIEW

File: pysixdesk/lib/dbtable.py [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+        return filemap
+
+    def customize_tables(self, table_names, info, dtype=None):
+        '''Customize the tables'''
+        if isinstance(info, dict):
+            for key, val in info.items():
+                self.tables[table_names][key] = self.type_dict[val]
+        elif isinstance(info, list):
+            if dtype is None:
+                raise Exception("Data type information needed!")
```
Comment (loiccoyle) [inline pysixdesk/lib/dbtable.py]: Raising a `ValueError` is more accurate, than the unnecessarily broad `Exception`.

File: pysixdesk/lib/dbtable.py [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+            for key in info:
+                self.tables[table_names][key] = dtype
+        else:
+            raise Exception("Unsupported input format!")
```
Comment (loiccoyle) [inline pysixdesk/lib/dbtable.py]: `Exception` --> `TypeError` and maybe change 'format' to 'type' ?

File: pysixdesk/lib/dbtable.py (hunk 1 above)
Comment (luxiaohan) [reply pysixdesk/lib/dbtable.py]: Make sense!

File: pysixdesk/lib/dbtable.py (hunk 2 above)
Comment (luxiaohan) [reply pysixdesk/lib/dbtable.py]: Make sense,  'type' is more accurate.

**Pull request merged**